### PR TITLE
Add perf related workflows.

### PR DESCRIPTION
**Pull request by jeremyjiang-dev**
Added quickstart and catalyst worflows.

### CODE REVIEW

**Comment by jeremyjiang-dev** [comment]
@paulb777 Hello Paul, I am trying to add quickstart workflow for perf, but I do not have the permission to add secret files. Could you help me to create a secret with `qs-performance.plist.gpg` in https://github.com/firebase/firebase-ios-sdk/tree/master/scripts/gha-encrypted ? Thank you!

**Review by zijianjoy — APPROVED** [review]

**Review by paulb777 — APPROVED** [review]

File: .github/workflows/performance.yml [hunk 1]
```diff
@@ -47,3 +47,37 @@ jobs:
       run: scripts/setup_bundler.sh
     - name: Build
       run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebasePerformance.podspec --skip-tests --platforms=${{ matrix.target }}
+
+  quickstart:
+    # Don't run on private repo unless it is a PR.
```
Comment (visumickey) [inline .github/workflows/performance.yml:52]: Lets remove this comment.

File: .github/workflows/performance.yml [hunk 2]
```diff
@@ -47,3 +47,37 @@ jobs:
...
+    - name: Install Secret FIREGSignInInfo.h
+      run: scripts/decrypt_gha_secret.sh scripts/gha-encrypted/FIREGSignInInfo.h.gpg \
+          quickstart-ios/TestUtils/FIREGSignInInfo.h "$signin_secret"
+    - name: Test swift quickstart
+      run: ([ -z $plist_secret ] || scripts/third_party/travis/retry.sh scripts/test_quickstart.sh Performance)
+    - name: Test objc quickstart
+      run: ([ -z $plist_secret ] || scripts/third_party/travis/retry.sh scripts/test_quickstart.sh Performance)
+
+  catalyst:
+    # Don't run on private repo unless it is a PR.
```
Comment (visumickey) [inline .github/workflows/performance.yml:75]: Lets remove this.

**Review by visumickey — APPROVED** [review]

File: .github/workflows/performance.yml (hunk 1 above)
Comment (jeremyjiang-dev) [reply .github/workflows/performance.yml]: Sure.Removed.

File: .github/workflows/performance.yml (hunk 2 above)
Comment (jeremyjiang-dev) [reply .github/workflows/performance.yml]: Sure. Removed.

**Pull request merged**